plugins/basic/python/tk_maya_basic/plugin_logging.py

```diff
@@ -0,0 +1,71 @@
+# Copyright (c) 2016 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import logging
+import time
+
+import maya.api.OpenMaya as OpenMaya2  # Python API 2.0
+import maya.utils
+
+
+class PluginLoggingHandler(logging.Handler):
+    """
+    Custom logging handler to display plug-in logging records in Maya script editor in a thread safe manner.
+    """
+
+    def __init__(self, plugin_name):
+        """
+        Initializes an instance of the plug-in logging handler.
+
+        :param plugin_name: Plug-in name to include in the standard message format.
+        """
+
+        super(PluginLoggingHandler, self).__init__()
+
+        # Set the handler to use a standard message format similar to the one used by the Maya engine.
+        logging_format = "%s: %s" % (plugin_name, "%(message)s")
+        self.setFormatter(logging.Formatter(logging_format))
+
+        # Time stamp used when logging debug messages.
+        # Initialized with the current time for lack of a better value.
+        self._last_debug_msg_time = time.time()
+
+    def emit(self, record):
+        """
+        Displays the specified logging record in Maya script editor according to its level number.
+
+        This method is the implementation override of the 'logging.Handler' base class one.
+
+        :param record: Logging record to display in Maya script editor.
+        """
+
+        # Give a standard format to the message.
+        msg = self.format(record)
+
+        # Display the message in Maya script editor in a thread safe manner.
+        # The overall message format is similar to the one used by the Maya engine.
+        if record.levelno < logging.INFO:
+            # Time stamp the messagde with the elapsed time since the last display debug call.
+            current_time = time.time()
+            msg = "Shotgun Debug [%0.3fs]: %s" % (current_time-self._last_debug_msg_time, msg)
+            maya.utils.executeInMainThreadWithResult(OpenMaya2.MGlobal.displayInfo, msg)
+            self._last_debug_msg_time = current_time
+
+        elif record.levelno < logging.WARNING:
```

Comment:
Since a `<` is used in the cascading `if`/`elif`, it is not really simplifiable in one or two lines, but I managed to streamline the code nonetheless. The message format is now similar to the engine new (non-legacy) message format.